.github/actions/add-commit-status/action.yml

```diff
@@ -0,0 +1,39 @@
+name: Add Commit Status
+description: Adds status to a commit
+inputs:
+  statusContext:
+    description: The context passed to `createCommitStatus`
+    required: true
+  state:
+    description: The state to assign to the commit
+    required: true
+runs:
+  using: composite
+  steps:
+    - name: Add commit status
+      uses: actions/github-script@v6
+      env:
+        STATE: ${{ inputs.state }}
+        STATUS_CONTEXT: ${{ inputs.statusContext }}
+      with:
+        script: |
+          const {head_sha, status, repository} = context.payload.workflow_run;
```

Comment:
yeah i agree, the only reason i did suggest it is because I usually always look for some javascript file to match the workflow name and for that to have the bulk of the logic for the action, but also i agree that's it's not necessary to have another file